cpp/tests/reductions/segmented_reduction_tests.cpp

```diff
@@ -770,6 +772,197 @@ TEST_F(SegmentedReductionTestUntyped, NonNullableInput)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*res, null_init_expect);
 }
 
+TEST_F(SegmentedReductionTestUntyped, Mean)
```

Comment:
Can we add tests for a couple more cases:
- Errors raised for invalid types (non-arithmetic inputs should raise, non-floating `data_type` should raise)
- A column with no nulls and `null_policy::EXCLUDE` to ensure that the `thrust::adjacent_difference` code path for computing valid counts is executed